extensions/transport/portability-transport-jettyrest/src/main/java/org/datatransferproject/transport/jettyrest/http/JettyTransport.java

```diff
@@ -116,6 +120,58 @@ public void registerServlet(String path, Servlet servletContainer) {
 
   private class JettyErrorHandler extends ErrorHandler {
 
+    /**
+     * Hack to intercept any non-"api/*" requests that make it to the API server. This
+     * includes:
+     * - "/" -> index.html
+     * - "/callback?token=foo" -> index.html?token=foo
+     *
+     * <p>This is NOT used in the demo-server since nginx handles these requests with
+     * the Angular client.
+     *
+     * <p>This IS used by the demo-google-deployment, as the Google Cloud Load Balancer
+     * currently used in that deployment can't do URL remaps - so we solve that by setting
+     * our default mapping to the API backends and redirecting those requests here.
+     * Otherwise "/" will map to our static bucket and display a directory listing rather
+     * than index.html.
+     *
+     * TODO(rtannenbaum): Move this somewhere better than the error handler
+     */
+    @Override
+    public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response)
+            throws IOException
+    {
+      logger.warn("handling non-'api/*' request: " + request.getServletPath()
+              + ". Do NOT do this in production environments!");
+      if (request.getServletPath().equals("/")) {
+        String redirect = request.getServerName() + "/index.html";
+        redirect = (request.isSecure() || !request.getServerName().equals("localhost") ? "https://" : "http://")
+                + redirect;
+        logger.info("redirecting to: " + redirect);
+        response.sendRedirect(redirect);
+      } else if (request.getServletPath().startsWith("/callback")) {
+        // This is a hack to redirect OAuth callbacks back to the frontend so creds may be processed there.
+        // This should only be done for demo purposes. Secure implementations may not want to allow creds
+        // to reach here (i.e. the API server). We do this here due to limitations with the load balancer
+        // used in demo-google-deployment (we cannot handle the redirect there with the Google Cloud Load
+        // Balancer currently).
+        String redirect = request.getServerName() + "/index.html?";
+        redirect = (request.isSecure() || !request.getServerName().equals("localhost") ? "https://" : "http://")
+                + redirect;
+        for (Map.Entry<String, String[]> header : request.getParameterMap().entrySet()) {
+          if (header.getKey().equals("code")
```

Comment:
Maybe we can pass all params along?

Reply:
Outdated, we don't need this hack in the API any more with the new "website" static bucket.